haystack/document_store/elasticsearch.py

```diff
@@ -687,21 +687,39 @@ def update_embeddings(self, retriever: BaseRetriever, index: Optional[str] = Non
 
         bulk(self.client, doc_updates, request_timeout=300, refresh=self.refresh_type)
 
-    def add_eval_data(self, filename: str, doc_index: str = "eval_document", label_index: str = "label"):
+    def add_eval_data(self, filename: str, doc_index: str = "eval_document", label_index: str = "label",
+                      batch_size: Optional[int] = None):
         """
         Adds a SQuAD-formatted file to the DocumentStore in order to be able to perform evaluation on it.
 
-        :param filename: Name of the file containing evaluation data
+        :param filename: Name of the file containing evaluation data (json or jsonl)
         :type filename: str
         :param doc_index: Elasticsearch index where evaluation documents should be stored
         :type doc_index: str
         :param label_index: Elasticsearch index where labeled questions should be stored
         :type label_index: str
         """
+        if filename.endswith(".json"):
```

Comment:
Just suggestion. By creating one function in `BaseDocumentStore` and moving this piece of code there will remove this duplicate code from these 3 document stores.

Reply:
Yes, this makes totally sense! thanks :)